src/ArraySlice.hpp

```diff
@@ -246,6 +246,55 @@ class ArraySlice
     return m_data == ptr;
   }
 
+  /**
+   * @brief Check if the slice is contiguous in memory
+   * @tparam _USD dummy template parameter equal to USD; do not replace
+   * @return @p true if represented slice is contiguous in memory
+   */
+  template< int _USD = USD >
+  LVARRAY_HOST_DEVICE inline CONSTEXPRFUNC
+  std::enable_if_t< ( _USD >= 0), bool >
+  isContiguous() const
+  {
+    bool rval = true;
+    for( int i = 0 ; i < NDIM ; ++i )
+    {
+      if( i == _USD ) continue;
+      INDEX_TYPE prod = 1;
+      for( int j = 0 ; j < NDIM ; ++j )
+      {
+        if( j != i ) prod *= m_dims[j];
+      }
+      rval &= (m_strides[i] <= prod);
+    }
+    return rval;
+  }
+
+  /**
+   * @brief Check if the slice is contiguous in memory
+   * @tparam USD_ dummy template parameter equal to USD; do not replace
+   * @return @p false, this overload is enabled for slices that
+   *         have already lost its unit stride dimension
+   */
+  template< int USD_ = USD >
+  LVARRAY_HOST_DEVICE inline CONSTEXPRFUNC
+  std::enable_if_t< (USD_ < 0), bool >
+  isContiguous() const
+  {
+    return false;
+  }
+
+  /**
+   * @brief Return a pointer to the values.
+   * @pre The slice must be contiguous
+   */
+  LVARRAY_HOST_DEVICE inline CONSTEXPRFUNC
+  T * data() const
+  {
+    LVARRAY_ASSERT( isContiguous() );
```

Comment:
Could you rename this to `dataIfContiguous` and the `LVARRAY_ASSERT` to `LVARRAY_ERROR_IF`?